### ISSUE TITLE
ManiaModLoader-Installer.exe has stopped working

### ISSUE THREAD

**Issue by ShadowOne333** [thesupersonic16/mania-mod-loader#5]
I get the titled message right after opening up the Installer.
It stays at "Searching for Sonic Mania..." for like 5-10 seconds and then I get the message.
My version of Sonic Mania is installed at Program Files (x86)/Sonic Mania in case that helps.

Also, one more question, is this app supposed to update Sonic Mania to the latest 1.03.0919 version or does it only update the Mod Loader?
If not, how would I go about updating the game without entering Steam itself?

**Comment by thesupersonic16** [comment]
This tool is hard to debug since It uses Steam and the registry, Like I will need to see your Steam config files and your registry. I'll try make a version that will dump some debugging info.

**Comment by thesupersonic16** [comment]
Please Run this with `debug` as a argument. e.g. `> ManiaModLoader-Installer.exe debug`

After that Send `Debug.log`

[ManiaModLoader-Installer.zip](https://github.com/thesupersonic16/mania-mod-loader/files/1486488/ManiaModLoader-Installer.zip)

**Comment by thesupersonic16** [comment]
You still have the issue?

**Comment by ShadowOne333** [comment]
Sorry, I tested it out but forgot to get back to you.
I downloaded the .exe from the .zip you posted, opened up CMD and ran:
`ManiaModLoader-Installer.exe debug`

But nothing happened.
The .exe opened up as it usually does and it crashed in the same way.
It could have been that I did something wrong though.

**Comment by thesupersonic16** [comment]
The Crash is still expected and It should create a log file, Can you send it?

**Comment by ShadowOne333** [comment]
Is the Debug.log file supposed to be created in the same directory as the .exe or in another one?
I couldn't find any Debug.log file inside the same path, so I was guessing it could have been made in another directory.
What path should it be created in?

**Comment by thesupersonic16** [comment]
The path where CMD is currently set to. usually where the exe is at.

**Comment by thesupersonic16** [comment]
You can type the command in the address bar too.
![github003](https://user-images.githubusercontent.com/25761885/33162845-bca79c84-d07f-11e7-8578-51b4c4e4420f.png)

**Comment by ShadowOne333** [comment]
Sorry for taking so long to reply, I've been quite busy these last days.
I tried running it again with the command in the address bar as you mention, but still I get no Debug.log.
I also tried searching around in the PC for said file but couldn't find one.

Do I need to have something installed in order for the Debug.log to appear properly?

**Comment by thesupersonic16** [comment]
Here is one with a batch script.

[ManiaModLoader-Installer.zip](https://github.com/thesupersonic16/mania-mod-loader/files/1511390/ManiaModLoader-Installer.zip)

**Comment by ShadowOne333** [comment]
Downloaded it and it's running in the background.
How long would it take for it to generate the Debug.log file?
I see that Windows Error Reporting opened up and it's Preparing to send the information, but it's been 5+ mins stuck at 1%, should I leave it running anyway?

**Comment by MegaSuperab** [comment]
I [think I] GOT IT!

Because your game is in the Programs folder, Windows is doing everything it can to protect it. When I downloaded the latest installer, it thought it was malware and removed it. I think what’s happening is the installer is detected as a virus and doesn’t allow it to write into the Programs folder, which of course blocks access to Sonic Mania. Since the Steam version works just fine I’m not entirely sure, but I tried my best.

I know this next part sounds like I’m a game developer or something, but I suggest you get an actual copy of Sonic Mania rather than a pirated one. It’ll put it in a proper folder.

If you don’t want to cough up $20, try reinstalling the pirated version to a user-accessible folder such as the Downloads folder. Because it’s not part of the OS, Windows will let the installer scan it. Even if you do this, I don’t think this mod manager will work on older versions of the game.
